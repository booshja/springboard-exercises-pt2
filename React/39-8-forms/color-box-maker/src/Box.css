.Box {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: flex-end;
    padding: 2rem;
}

.Box--display {
    display: inline;
}

.Box--removebtn {
    border: 2px solid red;
    border-radius: 10px;
    padding: 3px 5px;
    margin-left: 0.5rem;
    background-color: transparent;
    color: red;
    font-weight: bold;
    height: 1.5rem;
    width: 1.5rem;
    cursor: pointer;
    text-align: center;
}

.Box--removebtn:hover {
    background-color: red;
    color: white;
    font-weight: bold;
}
