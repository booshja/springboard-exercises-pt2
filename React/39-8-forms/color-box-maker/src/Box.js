import React from "react";
import "./Box.css";

const Box = ({ color, width, height, id, removeBox }) => {
    // this component should display a div with a background color, width and height based on the props passed to it.
    return (
        <div className="Box" key={id} id={id}>
            <div
                style={{
                    backgroundColor: color,
                    width: `${width}px`,
                    height: `${height}px`,
                }}
                className="Box--display"
            ></div>
            <button className="Box--removebtn" onClick={() => removeBox(id)}>
                X
            </button>
        </div>
    );
};

export default Box;
