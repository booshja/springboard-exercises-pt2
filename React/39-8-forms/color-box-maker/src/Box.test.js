import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", () => {
    render(<Box color="green" width={100} height={100} id={7} />);
});

it("matches snapshots", () => {
    const { asFragment } = render(
        <Box color="green" width={100} height={100} id={7} />
    );
    expect(asFragment()).toMatchSnapshot();
});
