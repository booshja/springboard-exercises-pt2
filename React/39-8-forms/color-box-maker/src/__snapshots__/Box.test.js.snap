// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<DocumentFragment>
  <div
    class="Box"
    id="7"
  >
    <div
      class="Box--display"
      style="background-color: green; width: 100px; height: 100px;"
    />
    <button
      class="Box--removebtn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
